src/zebra_puzzles/evaluation.py

```diff
@@ -0,0 +1,373 @@
+        )
+    except BadRequestError as e:
+        if "'temperature' is not supported" in str(e):
+            response = client.beta.chat.completions.parse(
+                messages=[{"role": "user", "content": prompt}],
+                model=model,
+                seed=42,
+                response_format=response_format,
+            )
+        else:
+            raise e
+
+    # Reformat response
+    try:
+        output = response_format.model_validate(response.choices[0].message.parsed)
+    except ValidationError as e:
+        print("response:\n", response)
+        print(
+            "\nresponse.choices[0].message.parsed:\n",
+            response.choices[0].message.parsed,
+        )
+        print()
+        raise e
+
+    return output
+
+
+def compute_metrics(
+    scores_all_types: list[np.ndarray], score_types: list[str], n_puzzles: int
+) -> dict[str, tuple]:
+    """Compute the metrics.
+
+    For each score type e.g. cell score, a dictionary of metrics is computed. This dictionary includes a string describing the rounded metrics.
+
+    Args:
+        scores_all_types: Tuple of scores as numpy arrays. Each element contains the scores for a specific score type.
+        score_types: List of score type names as strings.
+        n_puzzles: Number of puzzles as an integer.
+
+    Returns:
+        Metrics as a dictionary of with the score type as the key, and the values being a tuple of ndarrays. The tuple contains the rounded metrics for the score type and a string describing the metrics for the score type.
+
+    TODO: Add more metrics e.g. from sklearn.metrics
+    """
+    # Number of score types
+    n_metrics = len(score_types)
+
+    # Initialize metrics
+    mean_scores = np.zeros(n_metrics, dtype=float)
+    std_scores = np.zeros(n_metrics, dtype=float)
+    std_mean_scores = np.zeros(n_metrics, dtype=float)
+
+    # Initialize strings describing metrics for each score type
+    score_strings = np.zeros(n_metrics, dtype="U100")
+
+    for i, scores in enumerate(scores_all_types):
+        # Take the mean
+        mean_scores[i] = float(np.mean(scores))
+
+        # Take the standard deviation
+        std_scores[i] = float(np.std(scores, ddof=1))
+
+        # Compute the standard deviation of the mean
+        std_mean_scores[i] = std_scores[i] / np.sqrt(float(n_puzzles))
+
+        # Round to significant digits
+        std_scores[i] = np.format_float_positional(
+            std_scores[i], precision=1, fractional=False
+        )
+        std_mean_scores[i] = np.format_float_positional(
+            std_mean_scores[i], precision=1, fractional=False
+        )
+        mean_precision = len(str(std_mean_scores[i]).split(".")[1])
+        mean_scores[i] = np.format_float_positional(
+            mean_scores[i], precision=mean_precision, fractional=False
+        )
+
+        # Describe the score with a string
+        score_str = f"\tMean: {mean_scores[i]} ± {std_mean_scores[i]}"
```

Comment:
If we're computing a, say, 95% confidence interval here, we're missing a constant, no?

```suggestion
        score_str = f"\tMean: {mean_scores[i]} ± 1.96 * {std_mean_scores[i]}"
```

Reply:
I have added a note on this in score_str